controllers/helmchartinstall_controller.go

```diff
@@ -47,8 +52,23 @@ func (r *HelmChartInstallReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	ctx := context.Background()
 	_ = r.Log.WithValues("helmchartinstall", req.NamespacedName)
 
-	// your logic here
-	i := &helmv1alpha1.HelmChartInstall{}
+	// TODO NOTE the group, version, and kind would normally come from the
+	// stack configuration, and would be part of the configuration for the render
+	// callback
+	//
+	// We grab the claim as an unstructured so that we can have the same code handle
+	// arbitrary claim types. The types will be erased by this point, so if a stack
+	// author wants to validate the schema of a claim, they can do it by putting a
+	// schema in the CRD of the claim type so that the claim's schema is validated
+	// at the time that the object is accepted by the api server.
+	i := &unstructured.Unstructured{}
+	i.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Group:   "helm.samples.stacks.crossplane.io",
+			Version: "v1alpha1",
+			Kind:    "HelmChartInstall",
+		},
```

Comment:
It seems like either these GVK strings should be constants, of the GVK itself should be a constant-ish package level variable (since a struct cannot be a constant).